docs/reference/eql/syntax.asciidoc

```diff
@@ -14,21 +14,59 @@ experimental::[]
 [[eql-basic-syntax]]
 === Basic syntax
 
-EQL queries require an event type and a matching condition. The `where` keyword connects them.
+EQL queries require an event type and a matching condition. The `where` keyword
+connects them.
 
 [source,eql]
 ----
 event_type where condition
 ----
 
-For example, the following EQL query matches `process` events with a `process.name`
-field value of `svchost.exe`:
+For example, the following EQL query matches `process` events with a
+`process.name` field value of `svchost.exe`:
 
 [source,eql]
 ----
 process where process.name == "svchost.exe"
 ----
 
+[discrete]
+[[eql-syntax-event-types]]
+==== Event types
+
+In {es}, an event type is a valid, indexed value of the
+<<eql-required-fields,event type field>>. You can set the event type field using
+the `event_type_field` parameter of the EQL search API.
+
+[discrete]
+[[eql-syntax-match-any-event-type]]
+===== Match any event type
+
+To match events of any type, use the `any` keyword.
+
+For example, the following EQL query matches any event with a
+`network.protocol` field value of `http`:
+
+[source,eql]
+----
+any where network.protocol == "http"
+----
+
+[discrete]
+[[eql-syntax-event-types-starting-with-digits]]
+===== Event types starting with digits
+
+EQL queries cannot start with a digit. To match event types starting with a
```

Comment:
I think it's safe to drop this section, since we've never suggested anywhere else that this should be supported behavior.
There is an issue that was created for this (#51853), but I think that's more of a feature request than a bug.

My worry here is that the emphasis on digits could cause confusion. You also currently run into the same problem if there are hyphens or special characters in the event type.

One thing we could do, is reuse or merge this section with the previous mention that documents that don't have an "event type" and can't that be queried with the form `<event type> where <condition>`, can use `any where`. I think it's good that you show cases where `any where ...`  is convenient, and also when it's necessary.

Reply:
Thanks for the feedback @rw-access. With ea38a5698e6, I added a sentence to the previous section stating that `any where` can be used for documents without an event type field. Feels much less corner-case-y.